src/gui/DatabaseTabWidget.cpp

```diff
@@ -112,126 +97,116 @@ Database* DatabaseTabWidget::execNewDatabaseWizard()
             tr("The created database has no key or KDF, refusing to save it.\n"
                "This is definitely a bug, please report it to the developers."),
             QMessageBox::Ok, QMessageBox::Ok);
-        return nullptr;
+        return {};
     }
 
+    db->setInitialized(true);
     return db;
 }
 
 void DatabaseTabWidget::newDatabase()
 {
-    auto* db = execNewDatabaseWizard();
+    auto db = execNewDatabaseWizard();
     if (!db) {
         return;
     }
 
-    DatabaseManagerStruct dbStruct;
-    dbStruct.dbWidget = new DatabaseWidget(db, this);
-    insertDatabase(db, dbStruct);
-
-    if (!saveDatabaseAs(db)) {
-        // mark database as dirty if user canceled save dialog
-        emit db->modifiedImmediate();
-    }
+    addDatabaseTab(new DatabaseWidget(db, this));
+    db->markAsModified();
 }
 
 void DatabaseTabWidget::openDatabase()
 {
```

Comment:
Why remove the home dir? That was added as a PR.

Reply:
Supposing you mean 

    QString fileName = fileDialog()->getOpenFileName(this, tr("Open database"), QDir::homePath(), filter);

It's just utterly annoying when the open dialogue always opens in the home dir and does not remember my last location. There are only few things that annoy me more than file selection dialogues which don't keep track of the most recently selected folder.